locations/spiders/ashleyhomestore.py

```diff
@@ -103,14 +96,17 @@ def parse_ca_store(self, response):
 
         opening_hours = OpeningHours()
         for hour in hours:
-            hour = hour.strip()
-            day, open_close = re.search(pattern, hour).groups()
-            if open_close == "Closed":
+            try:
+                hour = hour.strip()
+                day, open_close = re.search(pattern, hour).groups()
+                if open_close == "Closed":
+                    continue
+                open_time, close_time = open_close.split('-')
+                opening_hours.add_range(day=day[:2],
+                                        open_time=datetime.datetime.strptime(open_time, '%I:%M%p').strftime('%H:%M'),
```

Comment:
Instead of parsing/formatting the time here you can pas in the `open_time` or `close_time` string and the parse format. The `add_range()` code will do the work of parsing and formatting for you.

Reply:
Oh, nice! This part of the code was in there from the original, so I didn't look at it before. Will update.